templates/product/details.html

```diff
@@ -117,7 +117,11 @@ <h2 class="product__info__price">
             <form id="product-form" role="form" class="product-form clearfix" method="post"
                   action="{% url 'product:add-to-cart' product_id=product.pk slug=product.get_slug %}" novalidate>
               {% csrf_token %}
-              {% bootstrap_field form.variant %}
+              {% if has_variants %}
+                {% bootstrap_field form.variant %}
+              {% else %}
+                <input type="hidden" value="{{ first_variant.pk }}" name="variant" id="id_variant" />
+              {% endif %}
```

Comment:
Shouldn't this whole `if has_variants ... else` block be abstracted by `bootstrap_field form.variant`?

Reply:
This could be done at the `ProductForm` level. This form is able to determine whether there is only one variant and change the widget of `variant` field to `HiddenInput`.